xbmc/addons/AddonDll.cpp

```diff
@@ -461,5 +398,31 @@ ADDON_STATUS CAddonDll::TransferSettings()
   return ADDON_STATUS_OK;
 }
 
+bool CAddonDll::CheckAPIVersion(int type)
+{
+  /* check the API version */
+  const char* kodiVersion = kodi::addon::GetTypeVersion(type);
+  const char* addonVersion = m_pDll->GetAddonTypeVersion(type);
+
+  if (AddonVersion(addonVersion) != AddonVersion(kodiVersion))
+  {
+    CLog::Log(LOGERROR, "Add-on '%s' is using an incompatible API version for type '%s'. Kodi API version = '%s', add-on API version '%s'",
+                            Name().c_str(),
+                            kodi::addon::GetTypeName(type),
+                            kodiVersion,
+                            addonVersion);
+
+    std::string heading = StringUtils::Format("%s: %s", TranslateType(Type(), true).c_str(), Name().c_str());
+    std::string text = StringUtils::Format(g_localizeStrings.Get(29803).c_str(),
+                                           kodi::addon::GetTypeName(type),
+                                           kodiVersion,
+                                           addonVersion);
+    CGUIDialogOK::ShowAndGetInput(CVariant{heading}, CVariant{text});
```

Comment:
Don't make gui calls from deep down in the addons system please. We shouldn't have random functions adding popous. Instead this should be in the gui code tied to some user action (or notification if it's a background task)